tests/run-pass/volatile.rs

```diff
@@ -0,0 +1,11 @@
+// related: #58645
+#![feature(core_intrinsics)]
+use std::intrinsics::{volatile_load, volatile_store};
+
+pub fn main() {
+    unsafe {
+        let mut i: isize = 1;
+        volatile_store(&mut i, 2);
+        assert_eq!(volatile_load(&i), 2);
+    }
```

Comment:
Could you add a final `assert_eq!(i, 2);`, just to make sure "normal" loads also see the right thing? Other than that, LGTM.

Reply:
Oh also you changed it back to a type that worked before. Please keep a non-scalar type, like a pair of two integers.